### PR TITLE
[Chore] Docker setup

### PR DESCRIPTION
**Pull request by kilemensi**
## Description

- [x] Switch to using `wait-for-postgres.sh` in development, and
- [x] Switch to using [official Dokku GitHub action](https://github.com/marketplace/actions/dokku) for deployment.

## Type of change

- [x] Chore (non-breaking change which does not add visible functionality but improves code quality)

## Screenshots

Screenshot from 2021-06-15 16-43-06

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

**Comment by esirK** [comment]
@kilemensi  seems like the `Error 137` is back

**Comment by kilemensi** [comment]
> @kilemensi seems like the `Error 137` is back

Yes @esirK ... and I remember someone told me they killed it dead!

**Comment by esirK** [comment]
> > @kilemensi seems like the `Error 137` is back
> 
> Yes @esirK ... and I remember someone told me they killed it dead!

Yeah.
But changes to docker file and docker compose might have re introduced it. Not sure which part exactly

**Comment by esirK** [comment]
@kilemensi 
Previous it was because of docker not being able to create files in the server which we fixed by adding default compose volumes

**Comment by kilemensi** [comment]
> > > @kilemensi seems like the `Error 137` is back
> > 
> > 
> > Yes @esirK ... and I remember someone told me they killed it dead!
> 
> Yeah.
> But changes to docker file and docker compose might have re introduced it. Not sure which part exactly

Could be @esirK but when PRs like https://github.com/CodeForAfrica/actNOW/pull/26 that don't touch Docker setup at all also fail with the same error, it makes you think there is something deeper going on.

**Comment by esirK** [comment]
> 
> Could be @esirK but when PRs like #26 that don't touch Docker setup at all also fail with the same error, it makes you think there is something deeper going on.

True. But rerunning the workflow seems to fix it. It looks like it's just specific to `PRs` from `dependabot`. Could it be the because of [this](https://github.blog/changelog/2021-02-19-github-actions-workflows-triggered-by-dependabot-prs-will-run-with-read-only-permissions/)?

**Comment by kilemensi** [comment]
> > Could be @esirK but when PRs like #26 that don't touch Docker setup at all also fail with the same error, it makes you think there is something deeper going on.
> 
> True. But rerunning the workflow seems to fix it. It looks like it's just specific to `PRs` from `dependabot`. Could it be the because of [this](https://github.blog/changelog/2021-02-19-github-actions-workflows-triggered-by-dependabot-prs-will-run-with-read-only-permissions/)?

Could be @esirK ... Lets keep an eye on that!

**Comment by kilemensi** [comment]
Looks like it's the waiting for Postgres part that causes issues @esirK . Do we know if there is a time limit?

**Comment by kilemensi** [comment]
> Looks like it's the waiting for Postgres part that causes issues @esirK . Do we know if there is a time limit?

Looks like there was a stray parameter "db" in the commands @esirK . Fixed now. Please review.

**Review by esirK — APPROVED** [review]
🚀

File: docker-compose.yml [hunk 1]
```diff
@@ -14,7 +14,7 @@ services:
       # Allow DB access from outside the container
       - "54321:5432"
     healthcheck:
-      test: ["CMD-SHELL", "pg_isready -U postgres"]
+      test: ["CMD-SHELL", "pg_isready -U actnow"]
```
Comment (KhadijaMahanga) [inline docker-compose.yml:17]: 💯

**Review by KhadijaMahanga — APPROVED** [review]
This looks good 🚀 

I have also been getting `services.app.build contains unsupported option: 'target'` and had to change docker-compose version to a higher version. Is anyone getting the same?

**Comment by kilemensi** [comment]
> This looks good rocket
> 
> I have also been getting `services.app.build contains unsupported option: 'target'` and had to change docker-compose version to a higher version. Is anyone getting the same?

`target` is a new-ish option @KhadijaMahanga ... added in file format [`3.4`](https://docs.docker.com/compose/compose-file/compose-versioning/#version-34), docker version `17.09.0`.

What version are you running?

**Comment by KhadijaMahanga** [comment]
@kilemensi yes, I noticed that. I am using `Docker version 18.09.5` so I had to manually edit the `docker-compse.yml` with  
`version: "3.4"` instead of just `version: "3"` for the error to disappear.

version is still "3" in this PR

**Comment by kilemensi** [comment]
> @kilemensi yes, I noticed that. I am using `Docker version 18.09.5` so I had to manually edit the `docker-compse.yml` with
> `version: "3.4"` instead of just `version: "3"` for the error to disappear.
> 
> version is still "3" in this PR

Nah, I'm not getting that error @KhadijaMahanga (but I'm docker version `20.10.7`)... As a matter of fact, `version` is supposed to be deprecated [nowadays](https://github.com/compose-spec/compose-spec/blob/master/spec.md#version-top-level-element)

**Pull request merged**